pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+            'ends': segmentInfo['Ends'],
+        });
+    }
+    return ret;
+};
+
+var gettingDetails = false;
+var getDetailsRequested = false;
+function getSubscriptionDetails() {
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    if (gettingDetails) {
+        getDetailsRequested = true;
+        return;
+    }
+    getDetailsRequested = false;
+    gettingDetails = true;
+    cockpit.spawn(['subscription-manager', 'list'],
+            { directory: '/', superuser: "try", environ: ['LC_ALL=C'] })
+        .done(function(output) {
+            client.subscriptionStatus.products = parseMultipleSubscriptions(output);
+            gettingDetails = false;
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        })
+        .fail(function(ex) {
+            client.subscriptionStatus.error = ex;
+            console.warn("Subscriptions [getSubscriptionDetails]: couldn't get details: " + ex);
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        });
+}
+
+client.registerSystem = function(subscriptionDetails) {
+    var deferred = $.Deferred();
+
+    var args = ['subscription-manager', 'register'];
+    if (subscriptionDetails.url != 'default')
+        args.push('--serverurl', subscriptionDetails.serverUrl);
+
+    // activation keys can't be used with auto-attach
+    if (subscriptionDetails.activationKeys)
+        args.push('--activationkey', subscriptionDetails.activationKeys);
+    else
+        args.push('--auto-attach');
+
+    if (subscriptionDetails.user || subscriptionDetails.password)
+        args.push('--username', subscriptionDetails.user, '--password', subscriptionDetails.password);
+
+    // proxy is optional
+    if (subscriptionDetails.proxy) {
+        args.push('--proxy', subscriptionDetails.proxyServer,
+                  '--proxyuser', subscriptionDetails.proxyUser,
+                  '--proxypass', subscriptionDetails.proxyPass);
+    }
+
+    // only pass org info if user provided it
+    if (subscriptionDetails.org)
+        args.push('--org', subscriptionDetails.org);
+
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    var process = cockpit.spawn(args, {
+        directory: '/',
+        superuser: "require",
+        environ: ['LC_ALL=C'],
+        err: "out"
+    });
+
+    // once we start this, we're in an unknown state
+    client.subscriptionStatus.products = [];
+    client.subscriptionStatus.status = undefined;
+
+    var promise;
+    var buffer = '';
+    process
+        .input('')
+        .stream(function(text) {
+            buffer += text;
+        })
+        .done(function(output) {
+            deferred.resolve();
+        })
+        .fail(function(ex) {
```

Comment:
The ex should be logged in all cases where we don't recognize the output. Otherwise this becomes hard to debug.